enclave/sgx/openssl_init.c

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/types.h>
+
+/* Refer to openssl/crypto.h for the original definitions */
+#define OE_OPENSSL_INIT_NO_LOAD_CONFIG 0x00000080L
+
+/* Forward declarion */
+typedef struct oe_openssl_init_settings OE_OPENSSL_INIT_SETTINGS;
+int OE_OPENSSL_init_crypto(
+    uint64_t opts,
+    const OE_OPENSSL_INIT_SETTINGS* settings);
+
+/*
+ * Weak implementation that prevents link-time errors when an enclave
+ * does not link against OpenSSL libraries.
+ */
+int OE_OPENSSL_init_crypto(
+    uint64_t opts,
+    const OE_OPENSSL_INIT_SETTINGS* settings)
+{
+    OE_UNUSED(opts);
+    OE_UNUSED(settings);
+    return 0;
+}
+OE_WEAK_ALIAS(OE_OPENSSL_init_crypto, OPENSSL_init_crypto);
```

Comment:
Consider renaming the function so that if an enclave does not use openssl,  then `nm -l enclave | grep openssl` return empty.